File: typescript/src/adapters/anthropic.ts
import { Message, Context } from '../core/models.js';

export interface AnthropicMessage {
  role: string;
  content: string;
}

export class AnthropicAdapter {
  static toMessages(context: Context): [AnthropicMessage[], string | null] {
    const systemMessages: string[] = [];
    const anthropicMessages: AnthropicMessage[] = [];

    for (const msg of context.messages) {
      if (msg.role === 'system') {
        systemMessages.push(msg.content);
      } else {
        anthropicMessages.push({
          role: msg.role,
          content: msg.content,
        });
      }
    }

    const system = systemMessages.length > 0 ? systemMessages.join('\n\n') : null;
    return [anthropicMessages, system];
  }
}

File: python/src/agent_context/adapters/anthropic.py
from __future__ import annotations

from agent_context.core.models import Context


class AnthropicAdapter:
    """Adapter for converting Context to Anthropic-compatible message format."""

    @staticmethod
    def to_messages(context: Context) -> tuple[list[dict], str | None]:
        """Convert a Context to Anthropic-compatible (messages, system) format.

        Extracts system messages into a single string and returns non-system
        messages as simple {role, content} dicts.
        """
        system_parts = []
        messages = []

        for msg in context.messages:
            if msg.role == "system":
                system_parts.append(msg.content)
            else:
                messages.append({
                    "role": msg.role,
                    "content": msg.content,
                })

        system = "\n\n".join(system_parts) if system_parts else None
        return messages, system
